File: test_get_version.py
#!/usr/bin/env python3

import sys
sys.path.insert(0, '../jelly-fpga-client-py')

from jelly_fpga_client.jelly_fpga_control import JellyFpgaControl

def main():
    target = sys.argv[1] if len(sys.argv) > 1 else "127.0.0.1:8051"
    print(f"Connecting to: {target}")
    
    # サーバーに接続
    client = JellyFpgaControl(target)
    
    # バージョンを取得
    version = client.get_version()
    print(f"Server version: {version}")

if __name__ == "__main__":
    main()

File: examples/test_get_version.rs
use std::env;

pub mod jelly_fpga_control {
    tonic::include_proto!("jelly_fpga_control");
}

use jelly_fpga_control::jelly_fpga_control_client::JellyFpgaControlClient;
use jelly_fpga_control::Empty;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    // コマンドライン引数でIPアドレスを取得（省略可能）
    let args: Vec<String> = env::args().collect();
    let target = if args.len() >= 2 {
        &args[1]
    } else {
        "127.0.0.1:8051" // デフォルト
    };
    
    println!("Connecting to: {}", target);
    
    // サーバーに接続
    let target_url = if target.starts_with("http://") {
        target.to_string()
    } else {
        format!("http://{}", target)
    };

    let mut client = JellyFpgaControlClient::connect(target_url).await?;
    
    // GetVersionを呼び出し
    let request = tonic::Request::new(Empty {});
    let response = client.get_version(request).await?;
    
    println!("Server version: {}", response.into_inner().version);
    
    Ok(())
}